src/metaopt/core/io/database/mongodb.py

```diff
@@ -16,12 +16,22 @@
     AbstractDB, DatabaseError, DuplicateKeyError)
 
 
+AUTH_FAILED_MESSAGES = [
+    "auth failed",
+    "Authentication failed."]
+
+DUPLICATE_KEY_MESSAGES = [
+    "duplicate key error"]
+
+
 def mongodb_exception_wrapper(method):
     """Convert pymongo exceptions to generic exception types defined in src.core.io.database.
 
     Current exception types converted:
     pymongo.errors.DuplicateKeyError -> DuplicateKeyError
-    pymongo.errors.BulkWriteError[DuplicateKeyError] -> DuplicateKeyError
+    pymongo.errors.BulkWriteError[DUPLICATE_KEY_MESSAGES] -> DuplicateKeyError
+    pymongo.errors.ConnectionFailure -> DatabaseError
+    pymongo.errors.OperationFailure(AUTH_FAILED_MESSAGES) -> DatabaseError
```

Comment:
OCD here... above it is with square brackets, here it is with paretheses... sorry for that :nerd_face:

Reply:
@bouthilx this one

Reply:
I would agree, but [] vs () has a meaning. BulkWriteError contains a list of errors, from which we convert the one with DUPLICATE_KEY_MESSAGES. OperationFailure has only a message.